app/javascript/components/App.js

```diff
@@ -75,6 +76,19 @@ export default class App extends Component {
         <Header {...this.props} />
         <Switch>
           <Route exact path="/" component={Home} />
+          <Route
+            path="/apartmentindex"
+            render={() => <ApartmentIndex apartments={this.state.apartments} />}
+          />
+          <Route
+            path="/apartmentprotectedindex"
+            render={() => (
+              <ApartmentProtectedIndex
+                apartments={this.state.apartments}
+                {...this.props}
```

Comment:
You may want to consider doing some JavaScript logic here to pass only the apartments that belong to the current user. It will reduce the about of content you are sending to your component.